Fitter/interface/CategoryManager.h

```diff
@@ -137,5 +138,21 @@ void CategoryManager::mergeCategories(TString mrg_name, std::vector<TString> cat
     this->all_cats[mrg_name.Data()] = merged_cat; // This might be a bit spicy...
 }
 
+
+void CategoryManager::renameCategory(TString new_name, TString old_name) {
+    if (this->hasCategory(new_name)) {
+        // Invalid new_name
+        std::cout << TString::Format("[WARNING] Unable to rename category. Category already exists: %s",new_name.Data()) << std::endl;
+        return;
+    }
+    if (!this->hasCategory(old_name)) {
+        std::cout << TString::Format("[WARNING] Unable to find category: %s",old_name.Data()) << std::endl;
+        return;
+    }
+    AnalysisCategory* cat = this->getCategory(old_name.Data());
+    cat->cat_name = new_name.Data();
+    this->all_cats[new_name.Data()] = cat;
+}
+
```

Comment:
While I'm not opposed to adding this method, you need to be a bit careful when doing this renaming. Specifically, the underlying `AnalysisCategory` objects use `AnalysisCategory::getName` (which returns `cat_name`) to refer to any children `AnalysisCategory`, i.e. those objects that were combined to create a merged `AnalysisCategory`.

I'm not sure if it would cause any problems in the current code, but if we ever tried to look up the children of a merged category using the `CategoryManager`, I'm pretty sure you would get an error, if you renamed any of those children categories at any point prior.

You _could_ try to find all the categories that your to-be-renamed category is a child of and then make sure to also change the corresponding name in _that_ (i.e. the parent's) category `.children` vector, but that seems complicated and unnecessarily computationally expensive, e.g. think of the (very unrealistic) case where you try to rename a category that is a child of >1000 other `AnalysisCategory`'s.

This approach (as well as you current implementation) also has the undesirable effect of modifying the underlying object's data. This is more or less harmless in the case where your wrapper class (in this case `CategoryManager`) is the sole 'owner' of the underlying data, but becomes a nightmare if you have multiple different objects accessing the same underlying data. Think of the case where you had two different `CategoryManager` instances for some reason and you called the `renameCategory` method on one of the managers for a category that is present in both managers. What would happen is that the name change would be reflected in both managers, however, in the case of the 2nd manager you would still need to use to _old name_ in order to access the category (since you didn't touch the ana cat dictionary of the 2nd manager), but when you called `getName()` on the ana cat returned by the 2nd manager you would get the _new name_, which could make for some very confusing errors down the line.

If you really want to add a method like this for the sake of convenience or something, then I would suggest that you instead implement an _alias_ mechanism of some kind, where the alias mapping (i.e. new <-> old) is stored entirely within the `CategoryManager` itself and then whenever you called `CategoryManager::getCategory` or `CategoryManager::hasCategory` it would try to use that mapping to find a category that you may have 'renamed'.

Although, I'm not a big fan of the aliasing option either, since you would still need to be figure out how you're going to manage the internal mapping of (potentially) multiple new namings to the same underlying old name that corresponds to the actually `AnalysisCategory`. Furthermore, this would still have the (potentially) unintuitive issue where `AnalysisCategory::getName` would be different than the name you passed to `CategoryManager::getCategory`.

Ultimately I think I would need to know what your specific use case here is. Typically the way I handled renaming things is by defining the string <-> string map in the overarching plotting macro that took care of the name replacements only when you went to actually fill in the names/labels of a given plot. After all, we could refer to the ttH process as "Beeblebrox" internally and then just make sure to replace the name when we go to generate the plot and no one would be any the wiser!